models/iaf_wang_2002.h

```diff
@@ -0,0 +1,518 @@
+/*
+ *  iaf_wang_2002.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IAF_WANG_2002_H
+#define IAF_WANG_2002_H
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// C includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+/**
+ * Function computing right-hand side of ODE for GSL solver.
+ * @note Must be declared here so we can befriend it in class.
+ * @note Must have C-linkage for passing to GSL. Internally, it is
+ *       a first-class C++ function, but cannot be a member function
+ *       because of the C-linkage.
+ * @note No point in declaring it inline, since it is called
+ *       through a function pointer.
+ * @param void* Pointer to model neuron instance.
+ */
+extern "C" inline int iaf_wang_2002_dynamics( double, const double*, double*, void* );
+
+
+/* BeginUserDocs: neuron, integrate-and-fire, conductance-based
+
+Short description
++++++++++++++++++
+
+Leaky integrate-and-fire-neuron model with conductance based synapses, and additional NMDA receptors.
+
+Description
++++++++++++
+
+``iaf_wang_2002`` is a leaky integrate-and-fire neuron model with
+
+* an approximate version of the neuron model described in [1]_.
+* exponential conductance-based AMPA and GABA-synapses
+* exponential conductance-based NMDA-synapses weighted such that it approximates the original non-linear dynamics
+* a fixed refractory period
+* no adaptation mechanisms
+
+Neuron and synaptic dynamics
+..................................................
+
+The membrane potential and synaptic variables evolve according to
+
+.. math::
+
```

Comment:
Some of the math code here seems off so it is not rendered. You can see the rendered result at https://nest-simulator--2979.org.readthedocs.build/en/2979/models/iaf_wang_2002.html

Building the docs on your computer should also emit LaTeX error messages.